src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy

```diff
@@ -418,16 +418,21 @@ class UnityPlugin implements Plugin<Project> {
             task.group = GROUP
             task.projectManifestFile.convention(extension.projectManifestFile)
             task.upmPackages.putAll(extension.upmPackages)
+            //needed in order to be able to generate solutions
+            task.upmPackages.put("com.unity.ide.rider", "3.0.28")
```

Comment:
Make the version perhaps a variable? Perhaps the same for the one right below too.

Reply:
WHere would I expose this variable? In the main extension? Sounds out of place there. This can already be overriden, just put another entry on `unity.upmPackages` map, for instance
```groovy
unity{
  upmPackages.put("com.unity.ide.rider", "other_version")
}

Reply:
I guess I can improve this to only run if a `GenerateSolution`-typed class is present, like
```groovy
project.gradle.taskGraph.whenReady {
                if(project.gradle.taskGraph.allTasks.any {it instanceof GenerateSolution}) {
                    task.upmPackages.put("com.unity.ide.rider", "3.0.28")
                }
            }
            ```

Reply:
Actually gave a whole rework in this region, please take another look.